webapp/src/cloudsync.ts

```diff
@@ -416,37 +514,51 @@ export function syncAsync(): Promise<void> {
     return updateNameAsync()
         .then(() => provider.listAsync())
         .then(entries => {
-            let allScripts = ws.getHeaders()
+            // Get all local headers including those that had been deleted
+            let allScripts = ws.getHeaders(true)
             let cloudHeaders = U.toDictionary(entries, e => e.id)
-            let existingHeaders = U.toDictionary(allScripts, h => h.blobId)
-            let waitFor = allScripts.map(hd => {
+            let existingHeaders = U.toDictionary(allScripts.filter(h => h.blobId), h => h.blobId)
+            console.log('all', allScripts);
```

Comment:
make logging optional?

Reply:
removing. note that this code path is only used by onedrive/google drive.

Reply:
svgoed